Blog/forms.py

```diff
@@ -16,4 +20,20 @@ def clean(self):
         title = cleaned_data.get("title")
         content = cleaned_data.get("content")
         if len(content) < len(title):
-            raise forms.ValidationError("content should be longer than title.")
\ No newline at end of file
+            raise forms.ValidationError("content should be longer than title.")
+
+    def clean_title(self):
+        title = self.cleaned_data.get("title")
+        soup = BeautifulSoup(title, features="html5lib")
+        for tag in soup.findAll(True):
+            if tag.name not in VALID_TAGS:
+                tag.hidden = True
+        return soup.renderContents().decode("utf-8")
+
+    def clean_content(self):
+        content = self.cleaned_data.get("content")
```

Comment:
Ensure illegal tags are escaped without deleting the content. If there is a <script> tag, allow it to become &lt;script&lt; so that the content will not disappear.